packages/terra-signature/src/Signature.jsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import './Signature.scss';
+
+const LineWidths = {
+  EXTRAFINE: 1,
+  FINE: 2,
+  MEDIUM: 4,
+  HEAVY: 6,
+};
+
+
+const propTypes = {
+  /**
+   * The line width to use when drawing the signature on the canvas.
+   * One of LineWidths.EXTRAFINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY.
+   */
+  lineWidth: PropTypes.oneOf([LineWidths.EXTRAFINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY]),
+
+  /**
+  * Line segments that define signature.
+  */
+  lineSegments: PropTypes.array,
+
+  /**
+   * A callback function to execute when a line segment is drawn. The first parameter is the event, the
+   * second parameter is all the line segments, and the last parameter is the most recent line segment drawn.
+   */
+  onChange: PropTypes.func,
+
+};
+
+const defaultProps = {
+  lineWidth: LineWidths.FINE,
+  lineSegments: undefined,
+  onChange: undefined,
+};
+
+class Signature extends React.Component {
+
+  constructor() {
+    super();
+
+    this.state = { lineSegments: [] };
+  }
+
+  componentDidMount() {
+    const canvas = this.node;
+
+    this.node.addEventListener('mousedown', e => this.mouseDown(e));
+    this.node.addEventListener('mousemove', e => this.mouseXY(e));
+    document.body.addEventListener('mouseup', e => this.mouseUp(e));
+
+    // For mobile
+    this.node.addEventListener('touchstart', e => this.mouseDown(e), false);
+    this.node.addEventListener('touchmove', e => this.mouseXY(e), true);
+    this.node.addEventListener('touchend', e => this.mouseUp(e), false);
+    document.body.addEventListener('touchcancel', e => this.mouseUp(e).bind(this), false);
+
+    this.node.width = canvas.getBoundingClientRect().width;
+    this.node.height = canvas.getBoundingClientRect().height;
+
+    const context = this.node.getContext('2d');
+
+    context.lineWidth = this.props.lineWidth;
+
+    const style = window.getComputedStyle(this.node);
+    const color = style.getPropertyValue('color');
+    context.strokeStyle = color;
+
+    if (this.props.lineSegments) {
+      this.state.lineSegments = this.props.lineSegments;
+      this.drawSignature(this.state.lineSegments);
+    }
+
+    window.addEventListener('resize', () => this.updateDimensions());
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.lineSegments !== nextProps.lineSegments) {
+      const newState = Object.assign({}, this.state, { lineSegments: nextProps.lineSegments });
```

Comment:
I don't think this line is needed. With only one state object you should be able to call:
```
this.setState({ lineSegments: nextProps.lineSegments })
```

Is the if statement needed? You could also simplify this a little more by deconstructing lineSegments